src/v/cloud_roles/request_response_helpers.cc

```diff
@@ -94,57 +115,49 @@ json::Document parse_json_response(iobuf resp) {
     return doc;
 }
 
+static ss::future<api_response> make_post_request(
+  http::client& client,
+  http::client::request_header& req,
+  iobuf&& content,
```

Comment:
please pass by value

Reply:
Sure, Since this is a very small change I added it to https://github.com/redpanda-data/redpanda/pull/7242 instead of making a PR for it.